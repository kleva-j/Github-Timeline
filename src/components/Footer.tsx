import React from 'react';

interface Props {
  absolute?: boolean;
}

export const Footer: React.FC<Props> = ({ absolute }) => {
  return (
    <>
      <footer
        data-testid="footer"
        className={(absolute ? 'absolute w-full bottom-0 bg-gray-900' : 'relative') + ' pb-6'}
      >
        <div className="container mx-auto px-4">
          <hr className="mb-6 border-b-1 border-gray-700" />
          <div className="flex flex-wrap items-center md:justify-between justify-center">
            <div className="w-full md:w-4/12 px-4">
              <div className="text-sm text-white font-semibold py-1">
                Copyright Â© {new Date().getFullYear()}{' '}
                <a
                  href="http://michaelobasi.com"
                  className="text-white hover:text-gray-400 text-sm font-semibold py-1"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Github Timeline
                </a>
              </div>
            </div>
            <div className="w-full md:w-8/12 px-4">
              <ul className="flex flex-wrap list-none md:justify-end  justify-center">
                <li>
                  <a
                    href="http://michaelobasi.com"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Michael Obasi
                  </a>
                </li>
                <li>
                  <a
                    href="http://michaelobasi.com"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    About Me
                  </a>
                </li>
                <li>
                  <a
                    href="https://frontendtips.netlify.app/"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Blog
                  </a>
                </li>
                <li>
                  <a
                    href="https://github.com/kleva-j/Github-Timeline/blob/develop/LICENCE"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    MIT License
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </footer>
    </>
  );
};
