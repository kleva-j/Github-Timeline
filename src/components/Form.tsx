import React from 'react';

export const SearchForms: React.FC = () => {
  return (
    <form className="flex flex-col sm:flex-row max-w-3xl mx-auto">
      <aside className="flex-1 my-2 md:mx-2 md:ml-0">
        <label htmlFor="desc" className="text-sm font-semibold">Job Description</label>
        <input
          id="desc"
          type="text"
          name="description"
          placeholder="Filter by title, benefits, companies, expertise"
          className="w-full py-2 px-4 border rounded bg-white"
        />
      </aside>

      <aside className="flex-1 my-2 sm:mx-2 md:mr-0">
        <label htmlFor="location" className="text-sm font-semibold">Location</label>
        <input
          type="text"
          id="location"
          name="location"
          placeholder="Filter by city, state, zip code or country"
          className="w-full py-2 px-4 border rounded bg-white"
        />
      </aside>
    </form>
  );
};
