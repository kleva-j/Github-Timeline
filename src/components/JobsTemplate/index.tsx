export * from './template1';
export * from './template2';
