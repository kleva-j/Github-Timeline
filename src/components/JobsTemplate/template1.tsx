import React from 'react';

import { Job } from 'components/SingleJob';

import { JobItem, TagSm } from 'styled';

export const Template1: React.FC<Job> = ({ id, title, company, created_at, url, type, location }) => {
  return (
    <JobItem data-testid={`job-${id}`}>
      <main className="self-stretch w-20 mr-5 border"></main>
      <aside>
        <div className="flex items-center">
          <h4 data-testid="title" className="text-base font-semibold truncate whitespace-nowrap text-gray-800">
            <a href={url} rel="noopener noreferrer" target="_blank">
              {title} {' - '}
            </a>
          </h4>
          <p data-testid="company" className="capitalize text-gray-600 text-sm pl-2 truncate whitespace-nowrap">
            {company}
          </p>
        </div>

        <p data-testid="created_at" className="text-gray-700 text-sm">
          {new Date(created_at).toLocaleDateString('en-GB', { dateStyle: 'short' })}
        </p>

        <div className="flex justify-start py-1">
          <TagSm data-testid="type" className="text-gray-700">
            {type}
          </TagSm>
          <TagSm data-testid="location" className="text-gray-700">
            {location}
          </TagSm>
        </div>
      </aside>
    </JobItem>
  );
};
