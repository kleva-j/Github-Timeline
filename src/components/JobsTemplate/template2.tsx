import React from 'react';
import PropTypes from 'prop-types';

import { Job } from 'components/SingleJob';

import { JobItem2, TagSm } from 'styled';
import defaultLogo from 'assets/svg/Company_logo.svg';

interface DateTimeFormatOptions {
  year?: 'numeric' | '2-digit';
  month?: 'numeric' | '2-digit' | 'long' | 'short' | 'narrow';
  day?: 'numeric' | '2-digit';
}

const options: DateTimeFormatOptions = {
  year: 'numeric',
  month: 'short',
  day: 'numeric',
};

const lang: string = navigator.language || 'en-US';

export const Template2: React.FC<Job> = ({
  id,
  title,
  company,
  created_at,
  url,
  type,
  location,
  company_url,
  company_logo,
}) => {
  const date = new Date(created_at);
  const formatedDate = new Intl.DateTimeFormat(lang, options).format(date);
  return (
    <JobItem2 data-testid={`job-${id}`}>
      <div className="flex flex-col sm:flex-row justify-between sm:items-center">
        <span className="font-light text-gray-600">{formatedDate}</span>
        <div className="flex justify-start py-1">
          <TagSm data-testid={`type-${id}`} className="text-gray-700 truncate whitespace-nowrap">
            {type}
          </TagSm>
          <TagSm data-testid={`location-${id}`} className="text-gray-700 truncate whitespace-nowrap">
            {location}
          </TagSm>
        </div>
      </div>
      <div className="mt-2" data-testid={`title-${id}`}>
        <a
          className="text-xl text-gray-700 font-bold hover:text-gray-600"
          href={url}
          rel="noopener noreferrer"
          target="_blank"
        >
          {title}
        </a>
      </div>
      <div className="flex flex-col sm:flex-row justify-between sm:items-center mt-4">
        <a
          className="text-blue-600 hover:underline whitespace-nowrap order-last sm:order-first mt-2 sm:mt-0"
          href={url}
          rel="noopener noreferrer"
          target="_blank"
        >
          Read more
        </a>
        <div>
          <a className="flex items-center" href={company_url} rel="noopener noreferrer" target="_blank">
            <img
              className="mx-4 w-10 h-10 object-cover hidden sm:block"
              src={company_logo || defaultLogo}
              alt="avatar"
            />
            <h1 className="text-gray-600 font-bold">{company}</h1>
          </a>
        </div>
      </div>
    </JobItem2>
  );
};

Template2.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  company: PropTypes.string.isRequired,
  created_at: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  company_url: PropTypes.string.isRequired,
  company_logo: PropTypes.string.isRequired,
  how_to_apply: PropTypes.string.isRequired,
};
