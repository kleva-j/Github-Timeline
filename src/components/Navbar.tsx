/* eslint-disable react/style-prop-object */
import React from 'react';
import { Link } from 'react-router-dom';

import Bars from 'assets/svg/Bars.svg';

interface Props {
  transparent?: boolean;
  navState?: boolean;
}

export const Navbar: React.FC<Props> = ({ transparent, navState = false }) => {
  const [navbarOpen, setNavbarOpen] = React.useState(navState);
  return (
    <nav
      data-testid="navbar"
      className={
        (transparent ? 'top-0 absolute z-50 w-full' : 'relative bg-white shadow-lg') +
        ' flex flex-wrap items-center justify-between px-2 py-3 navbar-expand-lg'
      }
    >
      <div className="container px-4 mx-auto flex flex-wrap items-center justify-between">
        <div className="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start">
          <Link
            className={
              (transparent ? 'text-white' : 'text-gray-800') +
              ' text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-no-wrap uppercase'
            }
            to="/"
          >
            Github Timeline
          </Link>
          <button
            className="cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-transparent rounded bg-transparent block lg:hidden outline-none focus:outline-none"
            type="button"
            onClick={() => setNavbarOpen(!navbarOpen)}
          >
            <i className={(transparent ? 'text-white' : 'text-gray-800') + ' fas fa-bars'}>
              <img src={Bars} alt="burger-nav" />
            </i>
          </button>
        </div>
        <div
          data-testid="list"
          className={
            'lg:flex flex-grow items-center bg-white lg:bg-transparent lg:shadow-none' +
            (navbarOpen ? ' block rounded shadow-lg' : ' hidden')
          }
          id="example-navbar-warning"
        >
          <ul className="flex flex-col lg:flex-row list-none mr-auto">
            <li className="flex items-center">
              <Link
                className={
                  (transparent
                    ? 'lg:text-white lg:hover:text-gray-300 text-gray-800'
                    : 'text-gray-800 hover:text-gray-600') +
                  ' px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold'
                }
                to="/jobs"
              >
                <i
                  className={
                    (transparent ? 'lg:text-gray-300 text-gray-500' : 'text-gray-500') +
                    ' far fa-file-alt text-lg leading-lg mr-2'
                  }
                />{' '}
                Jobs
              </Link>
            </li>
          </ul>
          <ul className="flex flex-col lg:flex-row list-none lg:ml-auto">
            <li className="flex items-center">
              <Link
                className={
                  (transparent
                    ? 'lg:text-white lg:hover:text-gray-300 text-gray-800'
                    : 'text-gray-800 hover:text-gray-600') +
                  ' px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold'
                }
                to="#twitter"
              >
                <i
                  className={
                    (transparent ? 'lg:text-gray-300 text-gray-500' : 'text-gray-500') +
                    ' fab fa-twitter text-lg leading-lg '
                  }
                />
                <span className="lg:hidden inline-block ml-2">Tweet</span>
              </Link>
            </li>

            <li className="flex items-center">
              <Link
                className={
                  (transparent
                    ? 'lg:text-white lg:hover:text-gray-300 text-gray-800'
                    : 'text-gray-800 hover:text-gray-600') +
                  ' px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold'
                }
                to="#github-stars"
              >
                <i
                  className={
                    (transparent ? 'lg:text-gray-300 text-gray-500' : 'text-gray-500') +
                    ' fab fa-github text-lg leading-lg '
                  }
                />
                <span className="lg:hidden inline-block ml-2">Star</span>
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
