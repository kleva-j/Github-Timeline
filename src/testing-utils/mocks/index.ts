import { LOAD_JOBS, LOAD_SINGLE_JOB } from 'GraphQL/Queries';

export const mockSingleJob = {
  request: {
    query: LOAD_SINGLE_JOB,
    variables: { id: 'XXXX' },
  },
  result: {
    data: {
      job: {
        id: 'XXXX',
        title: 'Buck',
        type: 'bulldog',
        company: 'company',
        description: 'description',
        url: 'some_random_url',
        location: 'location',
        created_at: 'Fri Mar 12 09:57:21 UTC 2021',
        company_url: 'some_company_url',
        company_logo: 'some_company_logo',
        how_to_apply: 'how_to_apply',
      },
    },
  },
};

export const errorMockSingleJob = {
  request: mockSingleJob.request,
  error: new Error('An error occurred'),
};

export const mockAllJobs = {
  request: {
    query: LOAD_JOBS,
    variables: {},
  },
  result: {
    data: {
      jobs: [
        {
          id: 'XXXX',
          title: 'Buck',
          type: 'bulldog',
          company: 'company',
          description: 'description',
          url: 'some_random_url',
          location: 'location',
          created_at: 'Fri Mar 12 09:57:21 UTC 2021',
          company_url: 'some_company_url',
          company_logo: 'some_company_logo',
          how_to_apply: 'how_to_apply',
        },
      ],
    },
  },
};

export const errorMockAllJobs = {
  request: mockAllJobs.request,
  error: new Error('An error occurred'),
};
