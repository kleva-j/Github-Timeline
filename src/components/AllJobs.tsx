import React from 'react';
import { useQuery } from '@apollo/client';

import { LOAD_JOBS } from 'GraphQL/Queries';

import { Job } from './SingleJob';
import { Template2 } from './JobsTemplate';

export const GetAllJobs: React.FC = () => {
  const { error, loading, data } = useQuery(LOAD_JOBS);

  if (loading)
    return (
      <p data-testid="loading" className="text-center my-16">
        Loading...
      </p>
    );
  if (error)
    return (
      <p data-testid="error" className="text-center my-16">
        Oops, Error occured!!!
      </p>
    );

  return (
    <article className="w-full" data-testid="jobs-article">
      {data?.jobs.slice(0, 10).map((item: Job) => (
        <Template2 {...item} key={item.id} />
      ))}
    </article>
  );  
};
