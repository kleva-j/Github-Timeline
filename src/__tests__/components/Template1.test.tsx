import { cleanup, render } from '@testing-library/react';

import { mockSingleJob } from 'testing-utils/mocks';

import { Template1 } from 'components/JobsTemplate';

describe('Template1', () => {
  afterEach(cleanup);

  const {
    result: {
      data: { job },
    },
  } = mockSingleJob;

  test('Should render the Template1 component.', () => {
    const { container, getByTestId } = render(<Template1 {...job} />);

    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('job-XXXX')).toBeTruthy();
  });
});
