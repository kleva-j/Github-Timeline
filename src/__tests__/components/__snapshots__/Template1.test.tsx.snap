// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template1 Should render the Template1 component. 1`] = `
<aside
  class="sc-hKgILt eXItxJ"
  data-testid="job-XXXX"
>
  <main
    class="self-stretch w-20 mr-5 border"
  />
  <aside>
    <div
      class="flex items-center"
    >
      <h4
        class="text-base font-semibold truncate whitespace-nowrap text-gray-800"
        data-testid="title"
      >
        <a
          href="some_random_url"
          rel="noopener noreferrer"
          target="_blank"
        >
          Buck
           
           - 
        </a>
      </h4>
      <p
        class="capitalize text-gray-600 text-sm pl-2 truncate whitespace-nowrap"
        data-testid="company"
      >
        company
      </p>
    </div>
    <p
      class="text-gray-700 text-sm"
      data-testid="created_at"
    >
      12/03/2021
    </p>
    <div
      class="flex justify-start py-1"
    >
      <button
        class="sc-iBPRYJ gEACbz text-gray-700"
        data-testid="type"
      >
        bulldog
      </button>
      <button
        class="sc-iBPRYJ gEACbz text-gray-700"
        data-testid="location"
      >
        location
      </button>
    </div>
  </aside>
</aside>
`;
